File: app/src/main/java/com/startandroid/data/Bookmarks.java
package com.startandroid.data;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.database.Cursor;

import static com.startandroid.data.Database.getDatabase;

public final class Bookmarks {
    public static Cursor getAllBookmarks() {
        return getDatabase().rawQuery("SELECT * FROM Bookmarks", null);
    }

    public static boolean add(int num, String title) {
        ContentValues cv = new ContentValues();
        cv.put("Number", num);
        cv.put("Title", title);
        long result = getDatabase().insert("Bookmarks", null, cv);
        return result != -1;
    }

    public static void remove(int num) {
        getDatabase().execSQL("DELETE FROM Bookmarks WHERE Number=" + num);
    }

    public static boolean isBookmarked(int num) {
        @SuppressLint("Recycle") Cursor cursor = getDatabase().rawQuery("SELECT Number FROM Bookmarks WHERE Number = " + num, null);
        return cursor.getCount() != 0;
    }
}

File: app/src/main/java/com/startandroid/data/Bookmarks.kt
package com.startandroid.data

import android.annotation.SuppressLint
import android.content.ContentValues
import android.database.Cursor

object Bookmarks {
    @JvmStatic
    val allBookmarks: Cursor
        get() = Database.getDatabase().rawQuery("SELECT * FROM Bookmarks", null)

    @JvmStatic
    fun add(num: Int, title: String?): Boolean {
        val cv = ContentValues().apply {
            put("Number", num)
            put("Title", title)
        }
        val result = Database.getDatabase().insert("Bookmarks", null, cv)
        return result != -1L
    }

    @JvmStatic
    fun remove(num: Int) {
        Database.getDatabase().execSQL("DELETE FROM Bookmarks WHERE Number=$num")
    }

    @JvmStatic
    fun isBookmarked(num: Int): Boolean {
        @SuppressLint("Recycle") val cursor = Database.getDatabase()
            .rawQuery("SELECT Number FROM Bookmarks WHERE Number = $num", null)
        return cursor.count != 0
    }
}
